src/components/Calculator.js

```diff
@@ -1,41 +1,76 @@
-import React from 'react';
+import React, { PureComponent, useState } from 'react';
+import calculate from '../logic/calculate';
 
-class Calculator extends React.PureComponent {
+class Calculator extends PureComponent {
   render() {
     return (
-      <div className="container">
-        <div className="result">
-          <input type="text" value={0} />
-        </div>
-        <div className="buttons">
-          <button type="button">AC</button>
-          <button type="button">+/-</button>
-          <button type="button">%</button>
-          <button type="button" className="opreator">፥</button>
-          <br />
-
-          <button type="button">7</button>
-          <button type="button">8</button>
-          <button type="button">9</button>
-          <button type="button" className="opreator">x</button>
-          <br />
-          <button type="button">4</button>
-          <button type="button">5</button>
-          <button type="button">6</button>
-          <button type="button" className="opreator">-</button>
-          <br />
-          <button type="button">1</button>
-          <button type="button">2</button>
-          <button type="button">3</button>
-          <button type="button" className="opreator">+</button>
-          <br />
-          <button type="button" className="zero"> 0 </button>
-          <button type="button">.</button>
-          <button type="button" className="opreator">=</button>
-          <br />
-        </div>
-      </div>
+      <Task />
     );
   }
 }
+const Task = () => {
+  const [obstate, setState] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
+
+  const btnOnClick = (e) => {
+    const show = () => calculate(obstate, e.target.name);
+    setState(show);
+  };
+
+  let result;
+  const { total, next, operation } = obstate;
+  if (total === null && next === null && operation === null) {
+    result = 0;
+  }
+  if (next !== null && total === null) {
+    result = next;
+  }
```

Comment:
- [ ] Great work on implementing the calculator logic, as stated in the requirements kindly use `state and props` by passing properties to child components from the `App.js` component
![image](https://user-images.githubusercontent.com/81932478/211975423-c0222f3d-15c3-49e8-b384-d744ab3f72b2.png)